Misc/NEWS.d/next/Security/2023-01-24-16-12-00.gh-issue-101283.9tqu39.rst

```diff
@@ -0,0 +1,6 @@
+Windows version of :meth:`subprocess.Popen.__init__` with
+``shell=True`` made its shell search algorithm protected from dropping
+a program named ``cmd.exe`` into a current directory. Now the current
+directory is used as the last resort if system environment variables
+are totally stripped and Windows is installed not into ``C\Windows``.
+Patch by Eryk Sun.
```

Comment:
```suggestion
On Windows, :meth:`subprocess.Popen.__init__` with ``shell=True`` 
now tries to use a full-qualified path for "cmd.exe" when ``ComSpec``
isn't defined. It looks for "cmd.exe" in its expected location 
relative to ``SystemRoot`` or ``SystemDrive`` in ``os.environ`` or, 
if neither is defined, relative to ``"C:\\Windows"``. This helps to 
protect against the system executing a malicious "cmd.exe" program
in the application directory or current working directory.  
Patch by Eryk Sun.
```

Reply:
This is the kind of too-much-detail I was concerned about in my review comments. Let's keep the NEWS entry simple and update the docs, but make sure we retain the ability to use an even more reliable algorithm in the future.